src/lib/utils/evm.ts

```diff
@@ -167,22 +170,49 @@ export const createBuildCustomApiTxInput = async (
   return { ...args, ...fees }
 }
 
-export const buildAndBroadcast = async ({ adapter, buildCustomTxInput }: BuildAndBroadcastArgs) => {
+export const buildAndBroadcast = async ({
+  adapter,
+  buildCustomTxInput,
+  receiverAddress,
+}: BuildAndBroadcastArgs) => {
+  const senderAddress = await adapter.getAddress(buildCustomTxInput)
```

Comment:
Prefer passing `{accountNumber, wallet}` from `buildCustomTxInput` explicitly, not to pass more properties than we need, but obviously fine